vent/io/devices/pins.py

```diff
@@ -82,25 +82,23 @@ class PWMOutput(Pin):
     _HARDWARE_PWM_PINS = (12, 13, 18, 19)
 
     def __init__(self, pin, initial_duty=0, frequency=None, pig=None):
-        """ Inherits attributes from parent Pin.
+        """ Inherits attributes from parent Pin, then sets PWM frequency & initial duty (use defaults if None)
 
         Args:
             pin (int): The number of the pin to use. Hardware PWM pins are 12, 13, 18, and 19.
             initial_duty (float): The initial duty cycle of the pin. Must be between 0 and 1.
             frequency (float): The PWM frequency to use.
             pig (PigpioConnection): pigpiod connection to use; if not specified, a new one is established
         """
-        super().__init__(pin, pig)
-        if self.pin not in self._HARDWARE_PWM_PINS:
-            self.hardware_enabled = False
-            frequency = self._DEFAULT_SOFT_FREQ if frequency is None else frequency
-            raise RuntimeWarning(
-                'PWMOutput called on pin {} but that is not a PWM channel.',
-                'Available frequencies will be limited.'.format(self.pin))
-        else:
-            self.hardware_enabled = True
-            frequency = self._DEFAULT_FREQUENCY if frequency is None else frequency
-        self.__pwm(frequency, initial_duty)
+        super().__init__(pin=pin, pig=pig)
+        self._hardware_enabled = True if self.pin in self._HARDWARE_PWM_PINS else False

```

Comment:
see this _hardware_enabled attr throughout, not quite sure what it signals

Reply:
The Raspberry Pi only supports hardware-generated PWM on pins 12, 13, 18, and 19, so generally `hardware_enabled` will be true if this is one of those, and false if it is not. However, `hardware_enabled` can also by dynamically set to False if for some reason pigpio is unable to start a hardware PWM (i.e. if the clock is unavailable or in use or something).

I'll add that ^^ in